### PR TITLE
Types List

### PR DESCRIPTION
**Pull request by Evilart86**
add TypeContainer
add TypesStore
add more info to displayed types

### CODE REVIEW

File: src/client/components/App.tsx [hunk 1]
```diff
@@ -2,56 +2,19 @@ import * as React from "react";
...
-        this.setState({
-            types: [...this.state.types, newType]
-        });
-    };
 
-    render() {
-        return (
-            <div>
-                <Navbar></Navbar>
+export default function App(props) {
```
Comment (aleh-rudzko) [inline src/client/components/App.tsx:8]: where extends React.Component ?

File: src/client/components/stores/TypesStore.ts [hunk 2]
```diff
@@ -1,24 +1,28 @@
...
-// }
 
-export default {};
\ No newline at end of file
+class TypesStore extends ReduceStore<Immutable.List<Type>, any> {
+    constructor() {
+	    super(TypeDispatcher);
+    }
+
+	getInitialState(): Immutable.List<Type> {
```
Comment (aleh-rudzko) [inline src/client/components/stores/TypesStore.ts]: pls, fix indentation
Also can we use just typescript features instead Immutable

File: src/client/components/stores/TypesStore.ts [hunk 3]
```diff
@@ -1,24 +1,28 @@
...
+		return Immutable.List<Type>();
+	}
+
+    reduce(state, action): Immutable.List<Type> {
+        switch (action.eventName) {
+            case TypeActionTypes.ADD_TYPE:
+		        return state.push(action.type);
```
Comment (aleh-rudzko) [inline src/client/components/stores/TypesStore.ts]: fix indentation and `return [...state.types, action.type]`

File: src/client/components/stores/TypesStore.ts [hunk 4]
```diff
@@ -1,24 +1,28 @@
...
 
-export default {};
\ No newline at end of file
+class TypesStore extends ReduceStore<Immutable.List<Type>, any> {
+    constructor() {
+	    super(TypeDispatcher);
+    }
+
+	getInitialState(): Immutable.List<Type> {
+		return Immutable.List<Type>();
```
Comment (aleh-rudzko) [inline src/client/components/stores/TypesStore.ts]: state should be a dict so pls return `{types: []}`

File: package.json [hunk 5]
```diff
@@ -3,6 +3,9 @@
   "version": "1.0.0",
   "description": "",
   "main": "index.js",
+  "engines": {
```
Comment (aleh-rudzko) [inline package.json:6]: what is it ?

**Review by aleh-rudzko — CHANGES_REQUESTED** [review]

File: src/client/components/App.tsx (hunk 1 above)
Comment (Evilart86) [reply src/client/components/App.tsx:8]: Did you sow how to make component-container in the Facebook repo?

Comment (aleh-rudzko) [reply src/client/components/App.tsx:8]: No, I didn't, can you explain why?

Comment (Evilart86) [reply src/client/components/App.tsx:8]: You can just make a view and pass all functional to it by Container.createFunctional.
All functional are our store and actions with which you can control the state of a view.

Comment (Evilart86) [reply src/client/components/App.tsx:8]: https://github.com/facebook/flux/tree/65724d5c0f08359f976a312c5581fdb99a799983/examples/flux-todomvc/src

Comment (aleh-rudzko) [reply src/client/components/App.tsx:8]: Ok, but you describe props where his interface?

File: package.json (hunk 5 above)
Comment (Evilart86) [reply package.json:6]: It needs to know what version of npm we used. For example i didn't know what version we are using until you told me.

Comment (Evilart86) [reply package.json:6]: https://docs.npmjs.com/files/package.json#engines

Comment (aleh-rudzko) [reply package.json:6]: you mean nodejs, ok, thx

File: src/client/components/stores/TypesStore.ts [hunk 6]
```diff
@@ -1,24 +1,27 @@
...
-// }
 
-export default {};
\ No newline at end of file
+class TypesStore extends ReduceStore<Type[], any> {
+    constructor() {
+	    super(TypeDispatcher);
+    }
+
+	getInitialState(): Type[] {
```
Comment (aleh-rudzko) [inline src/client/components/stores/TypesStore.ts]: pls, fix indentation

**Review by aleh-rudzko — CHANGES_REQUESTED** [review]

**Review by aleh-rudzko — APPROVED** [review]

**Pull request merged**